client.go

```diff
@@ -1423,6 +1486,101 @@ func (c *Client[TTx]) insertManyParams(params []InsertManyParams) ([]*riverdrive
 	return insertParams, nil
 }
 
+func (c *Client[TTx]) maybeNotifyInsert(ctx context.Context, execTx riverdriver.ExecutorTx, state rivertype.JobState, queue string) error {
+	if state != rivertype.JobStateAvailable {
+		return nil
+	}
+	return c.maybeNotifyInsertForQueues(ctx, execTx, []string{queue})
+}
+
+func (c *Client[TTx]) maybeNotifyInsertMany(ctx context.Context, tx riverdriver.ExecutorTx, insertParams []*riverdriver.JobInsertFastParams) error {
+	queues := make([]string, 0, len(insertParams))
```

Comment:
Small one, but thoughts on using a map instead of slice here? It probably doesn't matter for small numbers of insert many params, but for large ones (e.g. 1k or 10k like the benchmark does), you're allocating quite a large slice here unnecessarily since the cardinality of the queues in use is likely to be a tiny number like 1 or 2.

Reply:
Hmm, I'd have to then iterate through the map again afterward to get its keys, right? Maybe I should just scale back the allocation and allow it to reallocate as needed for larger slices? 🤔

Reply:
Opted to start with a default capacity of 10. This is probably going to be way more than enough for almost all River installs—how often are people really going to insert jobs on more than 10 different queues at one time?

Reply:
Yeah, should have relatively few total queues.

Looks good.